### PR TITLE
Baseline Package Template Files

### PR DESCRIPTION
**Pull request by Tomdye**
Initialising the basic package template

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{
+  "name": "dojo-cli",
+  "version": "0.0.3-pre",
```
Comment (agubler) [inline package.json]: version `2.0.0-pre` like all the others packages?

File: tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
+	"version": "2.0.0",
+	"compilerOptions": {
+		"declaration": false,
+		"module": "commonjs",
+		"noImplicitAny": true,
```
Comment (agubler) [inline tsconfig.json]: add `noImplicitThis` and `strictNullChecks`

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  global:
+  - BROWSERSTACK_USERNAME: dtktestaccount1
+  - BROWSERSTACK_ACCESS_KEY: mG2qbEFJCZY2qLsM7yfx
+  - SAUCE_USERNAME: dojo2-ts-ci
+  - SAUCE_ACCESS_KEY: e92610e3-834e-4bec-a3b5-6f7b9d874601
+cache:
+  directories:
+  - node_modules
+install:
+- "travis_retry npm install grunt-cli $(node -e \"var deps = require('./package.json').peerDependencies; for(var name in deps) process.stdout.write(name + '@' + deps[name] + ' ');\")"
```
Comment (agubler) [inline .travis.yml]: make this `travis_retry npm install grunt cli`

File: package.json [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  "keywords": [
+    "dojo",
+    "cli"
+  ],
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/dojo/cli.git"
+  },
+  "scripts": {
+    "prepublish": "grunt dist",
```
Comment (agubler) [inline package.json]: update `prepublish` task to `grunt peerDepInstall dist`

File: .travis.yml (hunk 3 above)
Comment (Tomdye) [inline .travis.yml]: with the other bit after it? or just change the first part?

File: package.json (hunk 4 above)
Comment (Tomdye) [inline package.json]: 👍

File: .travis.yml (hunk 3 above)
Comment (agubler) [inline .travis.yml]: no literally just `travis_retry npm install grunt cli`, see PR that is open to update this https://github.com/dojo/dojo2-package-template/pull/53

File: tsconfig.json (hunk 2 above)
Comment (Tomdye) [inline tsconfig.json]: done

**Comment by agubler** [comment]
👍

**Pull request merged**